Common/Ubuntu/ap_camera_non_gps/feature_match.py

```diff
@@ -0,0 +1,182 @@
+import numpy as np
+import cv2 as cv
+from matplotlib import pyplot as plt
+import subprocess
+import pyexiv2, json
+import math
+
+x_offset = 0
+y_offset = 0
+
+MIN_MATCH_COUNT = 30
+
+LOCATION_SCALING_FACTOR_INV = 89.83204953368922
+
+inCMX = 0
+inCMY = 0
+
+focal = 476.7030836014194
+
+
+
+def readExif(image):
+    with open(image, 'rb') as f:
+        with pyexiv2.ImageData(f.read()) as img:
+            data = img.read_exif()
+            print(data['Exif.Photo.UserComment'])
+    f.close()
+
+def modifyExif(image,data=None):
+    with open(image, 'rb+') as f:
+        with pyexiv2.ImageData(f.read()) as img:
+            matadata = data
+            dict1 = {'Exif.Photo.UserComment': json.dumps(metadata)}
+            img.modify_exif(dict1)
+            
+            f.seek(0)
+            f.truncate()
+            f.write(img.get_bytes())
+        f.seek(0)
+        with pyexiv2.ImageData(f.read()) as img:
+            result = img.read_exif()
+            #print(result)
+    f.close()
+
+
+def compParam(img1,img2):
+    global inCMY, inCMX
+    global x_offset, y_offset
+    img1 = cv.imread(img1,0)
+    img2 = cv.imread(img2,0)
+    #img2 = cv.resize(img2, (1778,866))
+    sift = cv.SIFT_create(nfeatures = 10000)
+
+    kp1, des1 = sift.detectAndCompute(img1,None)
+    kp2, des2 = sift.detectAndCompute(img2,None)
+    FLANN_INDEX_KDTREE = 1
+    index_params = dict(algorithm = FLANN_INDEX_KDTREE, trees = 5)
+    search_params = dict(checks = 50)
+    flann = cv.FlannBasedMatcher(index_params, search_params)
+    matches = flann.knnMatch(des1,des2,k=2)
+
+    good = []
+    for m,n in matches:
+        if m.distance < 0.7*n.distance:
+            good.append(m)
+
+
+    if len(good)>MIN_MATCH_COUNT:
+        src_pts = np.float32([ kp1[m.queryIdx].pt for m in good ]).reshape(-1,1,2)
+        dst_pts = np.float32([ kp2[m.trainIdx].pt for m in good ]).reshape(-1,1,2)
+        M, mask = cv.findHomography(src_pts, dst_pts, cv.RANSAC,5.0)
```

Comment:
For finding homography, since the images are in quick succession and not arbitrarily different transforms, are there any algorithms that may be optimized based on an initial guess?

Reply:
@Ryanf55 I am not sure if it is what you are mentioning, but Iterative Closest Point algorithm can serve the purpose. If it is required, I'll add it in the main algorithm